packages/content-management/table_list/src/reducer.tsx

```diff
@@ -0,0 +1,153 @@
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React from 'react';
+import { sortBy } from 'lodash';
+import { i18n } from '@kbn/i18n';
+
+import { UpdatedAtField } from './components';
+import type { State, UserContentCommonSchema } from './table_list_view';
+import type { Action } from './actions';
+import type { Services } from './services';
+
+interface Dependencies {
+  DateFormatterComp: Services['DateFormatterComp'];
+}
+
+function onInitialItemsFetch<T extends UserContentCommonSchema>(
+  items: T[],
+  { DateFormatterComp }: Dependencies
+) {
+  // We check if the saved object have the "updatedAt" metadata
+  // to render or not that column in the table
+  const hasUpdatedAtMetadata = Boolean(items.find((item) => Boolean(item.updatedAt)));
+
+  if (hasUpdatedAtMetadata) {
+    // Add "Last update" column and sort by that column initially
+    return {
+      tableSort: {
+        field: 'updatedAt' as keyof T,
+        direction: 'desc' as const,
+      },
+      tableColumns: [
+        {
+          field: 'updatedAt',
+          name: i18n.translate('contentManagement.tableList.lastUpdatedColumnTitle', {
+            defaultMessage: 'Last updated',
+          }),
+          render: (field: string, record: { updatedAt?: string }) => (
+            <UpdatedAtField dateTime={record.updatedAt} DateFormatterComp={DateFormatterComp} />
+          ),
+          sortable: true,
+          width: '150px',
+        },
+      ],
+    };
+  }
+
+  return {};
+}
+
+export function getReducer<T extends UserContentCommonSchema>({ DateFormatterComp }: Dependencies) {
+  return (state: State<T>, action: Action<T>): State<T> => {
+    switch (action.type) {
+      case 'onFetchItems': {
+        return {
+          ...state,
+          isFetchingItems: true,
+        };
+      }
+      case 'onFetchItemsSuccess': {
+        const items = action.data.response.hits;
+        // We only get the state on the initial fetch of items
+        // After that we don't want to reset the columns or change the sort after fetching
+        const { tableColumns, tableSort } = state.hasInitialFetchReturned
+          ? { tableColumns: undefined, tableSort: undefined }
+          : onInitialItemsFetch(items, { DateFormatterComp });
+
+        return {
+          ...state,
+          hasInitialFetchReturned: true,
+          isFetchingItems: false,
+          items: !state.searchQuery ? sortBy<T>(items, 'title') : items,
+          totalItems: action.data.response.total,
+          tableColumns: tableColumns
+            ? [...state.tableColumns, ...tableColumns]
+            : state.tableColumns,
+          tableSort: tableSort ?? state.tableSort,
+          pagination: {
+            ...state.pagination,
+            totalItemCount: items.length,
+          },
+        };
+      }
+      case 'onFetchItemsError': {
+        return {
+          ...state,
+          hasInitialFetchReturned: true,
```

Comment:
this approach with `hasInitialFetchReturned` seems quite brittle. But that aside, if the error was returned, is it ok to set this to `true`?

Reply:
Good catch, I removed it in https://github.com/elastic/kibana/pull/139556/commits/ec13757aade0ba3ac106ced8b5ee681a4d06ea63

Can you elaborate why you think it is brittle?